### PR TITLE
content: Lesson 3

### PR DESCRIPTION
**Pull request by agluszak**

### CODE REVIEW

**Comment by agluszak** [comment]
It's not ready for review yet @StarostaGit, I clicked that button involuntarily :P

File: content/lessons/4_enums/index.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+title = "Enums"
+date = 2022-03-13
+weight = 1
+[extra]
+lesson_date = 2022-03-14
++++
+
+## Enums
+
+It is often the case that you want to define a variable that can only take
```
[the remark this answers is not in the document]
Comment (agluszak) [reply content/lessons/4_enums/index.md:11]: "It is often the case" is weird wording? :o

I'll change it to first person

**Review by StarostaGit — APPROVED** [review]
Left some minor comments, but otherwise looks good.

File: content/lessons/4_enums/index.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+However, in C enums are just integers. Nothing prevents us from writing
+
+```c
+int main() {
+    enum shirt_size my_size = 666;
+    print_size(my_size);
+}
+```
+
+In C++ enums are more type-safe:
```
[the remark this answers is not in the document]
Comment (agluszak) [reply content/lessons/4_enums/index.md:25]: more type-safe than in C

**Comment by StarostaGit** [comment]
One thing, an example of `if let ...` would be useful.
We could consider adding the `func()?.method()` chaining syntax to the example as well, but we can also just mention it.

File: content/lessons/4_enums/index.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (StarostaGit) [reply content/lessons/4_enums/index.md:25]: I mean, no, enum class is type safe, period

File: content/lessons/4_enums/index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (StarostaGit) [reply content/lessons/4_enums/index.md]: I mean, not that it's incorrect, I guess just stylistically didn't sit right with me.

**Pull request merged**